### PR TITLE
Bump to iOS 14+

### PR DESCRIPTION
**Pull request by scannillo**
### Summary of changes

 These changes are in-line with our [Client Deprecation Policy](https://developer.paypal.com/braintree/docs/guides/client-sdk/deprecation-policy/ios/v5) which guarantees support for the current iOS version minus 2 (in this case since 16 is the latest, it guarantees 14,15, & 16).

- Update Package.swift, PopupBridge.podspec, & xcodeproject file:
   - Require Xcode 14.2
   - Require iOS 14+

 ### Checklist

 - [X] Added a changelog entry

### Authors
@scannillo

### CODE REVIEW

File: PopupBridge.podspec [hunk 1]
```diff
@@ -18,12 +18,8 @@ Use cases for PopupBridge:
...
-  s.ios.deployment_target = '9.0'
-  s.swift_version    = "5.7"
+  s.ios.deployment_target = '14.0'
+  s.swift_version    = "5.8"
   
   s.source_files = 'Sources/PopupBridge/**/*.swift'
-
-  # https://github.com/CocoaPods/CocoaPods/issues/10065#issuecomment-694266259
-  s.pod_target_xcconfig = { 'EXCLUDED_ARCHS[sdk=iphonesimulator*]' => 'arm64' }
-  s.user_target_xcconfig = { 'EXCLUDED_ARCHS[sdk=iphonesimulator*]' => 'arm64' }
```
Comment (scannillo) [inline PopupBridge.podspec:28]: This was required for Xcode 12

**Comment by scannillo** [comment]
So sad, GH actions doesn't have the Ventura (or Xcode 14.3) images yet 😢 https://github.com/actions/runner-images/issues/6426

File: README.md [hunk 2]
```diff
@@ -12,7 +12,9 @@ See the [Frequently Asked Questions](#frequently-asked-questions) to learn more
 Requirements
 ------------
 
-- iOS 9.0+
+- iOS 14.0+
+- Xcode 14.2+
+- Swift 5.7+, or Objective-C
```
Comment (jaxdesmarais) [inline README.md:17]: In the CHANGELOG we say `Xcode 14.3+ and Swift 5.8+` and here we have `Xcode 14.2+ and Swift 5.7+`. We should likely align the two.

**Review by sshropshire — APPROVED** [review]
🌊

File: README.md (hunk 2 above)
Comment (scannillo) [reply README.md:17]: Ah yes, I wanted to require 14.3+, but that requires a CI update. Will leave that for a separate PR. Will also update these to match.

**Pull request merged**